### PR TITLE
Estimated realized inflation

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

Estimated realized inflation calculation ([ref](https://github.com/AstarNetwork/astar-apps/issues/1259)). The calculation calculates slope from first block of the current cycle till the current block and assumes the same slope until the end of the period. This estimation introduces an error which will decrease after stakers clam their rewards (after first build and earn subperiod) and decrease even more as we are approaching the end of the current cycle.

We have beens struggling how to display inflation with just one meaningful number and seems this is the best way so far. In the next iteration we plan to add inflation chart as defined in the document referenced above.

 
image

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

### CODE REVIEW

File: src/hooks/useInflation.ts [hunk 1]
```diff
@@ -1,22 +1,106 @@
...
+    }
+
+    return inflation;
+  };
+
+  watch(
+    [$api],
+    async () => {
+      estimatedInflation.value = await estimateRealizedInflation();
+      console.log('Realized inflation:', estimatedInflation.value);
```
Comment (impelcrypto) [inline src/hooks/useInflation.ts:96]: Let's use `console.info` if you want to show this in the console

Comment (bobo-k2) [reply src/hooks/useInflation.ts]: Actually, this is not needed. Removed in https://github.com/AstarNetwork/astar-apps/pull/1280/commits/e120bdb75d843a397f4821a7f6ebf1ba47b229e8

File: src/hooks/useInflation.ts [hunk 2]
```diff
@@ -1,22 +1,105 @@
...
+   */
+  const estimateRealizedInflation = async (): Promise<number | undefined> => {
+    let inflation: number | undefined;
+
+    if ($api) {
+      try {
+        // Find the block when last NewInflationConfiguration event was emitted.
+        const subscanRepository = container.get<ISubscanRepository>(Symbols.SubscanRepository);
+        const response = await subscanRepository.getEvents(
+          networkNameSubstrate.value.toLocaleLowerCase(),
```
Comment (gtg7784) [inline src/hooks/useInflation.ts:42]: I think we need to use toLowerCase for this case

**Review by gtg7784 — CHANGES_REQUESTED** [review]

**Review by impelcrypto — APPROVED** [review]

**Review by gtg7784 — APPROVED** [review]
Thank you! LGTM

**Pull request merged**